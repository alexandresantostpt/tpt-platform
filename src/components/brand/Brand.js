import { config } from '@config'

import { events } from '@/constants/events'

import PubSub from 'pubsub-js'
import React, { useEffect, useState } from 'react'
import styled from 'styled-components'

import { getBrand } from '@utils/auth'

const StyledBrand = styled.span`
    display: inline-block;
    height: 3rem;
    ${({ image }) => {
        if (image) {
            return `
                background: url(${config.api.url}/agency/download/${image}) no-repeat center;
                background-size: contain;
            `
        }
    }};
    width: 10rem;
`

const Brand = () => {
    const [image, updateImage] = useState(getBrand())

    useEffect(() => {
        PubSub.subscribe(events.CHANGE_BRAND, (_, newImage) => updateImage(newImage))
    }, [])

    return <StyledBrand image={image} />
}

export default Brand
