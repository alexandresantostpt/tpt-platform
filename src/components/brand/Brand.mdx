---
name: Brand
route: /brand
---
import { Playground, PropsTable } from 'docz'

import Brand from './Brand'

import '@cfg/scripts'
import '@cfg/style'

# Brand
<PropsTable of={ Brand }/>

## Basic Usage

```
<Brand/>
```

<Playground>
    <Brand/>
</Playground>
