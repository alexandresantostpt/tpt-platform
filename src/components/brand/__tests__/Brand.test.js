import React from 'react'
import renderer from 'react-test-renderer'

import Brand from '@components/brand/Brand'

describe('Tests for Brand component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer.create(<Brand />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
