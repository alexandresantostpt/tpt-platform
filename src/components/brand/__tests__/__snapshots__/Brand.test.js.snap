// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Brand component Should rendering without crash  1`] = `
<span
  className="Brand__StyledBrand-sc-19fy8m0-0 jmSzTQ"
/>
`;
